### ISSUE TITLE
save() for emails is altering the body of the message

### ISSUE THREAD

**Issue by shellycao** [ecederstrand/exchangelib#38]
After calling save() on an email message, the body of the email changes to the html plain text version of the email. This can only be seen online though, since calling ```print(message.body)``` will give the same result either way.

Code:
```
from exchangelib import DELEGATE, Account, Credentials, Configuration, NTLM
creds = Credentials(
    username = 'abc',
    password = 'abc')
config = Configuration(
    server = 'abc.com',
    credentials = creds,
    auth_type = NTLM)
account = Account(
    primary_smtp_address = 'abc@abc.com',
    config = config,
    access_type = DELEGATE)
for message in account.inbox.all():
    message.save()
```

Before running the code:
![beforesave](https://cloud.githubusercontent.com/assets/16722945/20493365/7390e8ee-afe6-11e6-9004-bcb854490b60.PNG)

After (the screenshot cuts out the bottom part, but it is equivalent to calling ```message.body```):
![aftersave](https://cloud.githubusercontent.com/assets/16722945/20493377/7ec1e1a0-afe6-11e6-82e2-0541c6ef7d44.PNG)

**Comment by ecederstrand** [comment]
This is most likely because we unconditionally set `BodyType='Text'` in https://github.com/ecederstrand/exchangelib/blob/master/exchangelib/folders.py#L1039 but your body was received by `exchangelib` with `BodyType='HTML'`.

There's further discussion on HTML and text body in http://stackoverflow.com/questions/20982851/how-to-get-the-email-body-in-html-and-text-from-exchange-using-ews-in-c

Body element: https://msdn.microsoft.com/en-us/library/office/jj219983(v=exchg.150).aspx
TextBody element: https://msdn.microsoft.com/en-us/library/office/jj191321(v=exchg.150).aspx

Currently, `exchangelib` only supports text body. We need to find a reasonable way to retrieve both versions and specify if the body is HTML or text when saving. Maybe something like this:

```python
item = my_folder.all()[0]
item.body  # Contains the text version
item.body_as_html # Contains the HTML version

# Sends the HTML version if item.body_as_html is not empty, otherwise sends the text version. It may be possible to send both versions at once - I haven't tested
item.save()
```

**Comment by shellycao** [comment]
Okay thanks! I wasn't aware that this was a known issue.

**Comment by ecederstrand** [comment]
It's not, apart from a TODO deep in the code somewhere. Thanks for hitting the bug :-)

**Comment by ecederstrand** [comment]
I would prefer to keep it open, to track the issue. It still needs to be fixed

**Comment by ecederstrand** [comment]
This should be fixed now. I just published version 1.7.2  on PyPI with the changes
